src/test/java/cs2103/v15_1j/jimjim/DataListsTest.java

```diff
@@ -0,0 +1,154 @@
+import java.util.ArrayList;
+import java.util.List;
+
+import org.hamcrest.core.IsEqual;
+import org.hamcrest.core.IsNot;
+import org.junit.Before;
+import org.junit.Test;
+
+import cs2103.v15_1j.jimjim.model.DataLists;
+import cs2103.v15_1j.jimjim.model.DeadlineTask;
+import cs2103.v15_1j.jimjim.model.Event;
+import cs2103.v15_1j.jimjim.model.FloatingTask;
+
+public class DataListsTest {
+
+	DataLists displayList;
+	DataLists masterList;
+
+	@Before
+	public void setUp() throws Exception {
+		this.masterList = new DataLists();
+		this.displayList = new DataLists();
+	}
+
+	@Test
+	public void overloadedConstructorTest() {
+		Event event1 = new Event("Event 1", LocalDateTime.of(2016, Month.MARCH, 30, 12, 00), 
+				LocalDateTime.of(2016, Month.MARCH, 30, 13, 00));
+		Event event2 = new Event("Event 2", LocalDateTime.of(2016, Month.MARCH, 31, 12, 00), 
+				LocalDateTime.of(2016, Month.MARCH, 30, 13, 00));
+		FloatingTask fTask1 = new FloatingTask("Task 1");
+		FloatingTask fTask2 = new FloatingTask("Task 2");
+		FloatingTask fTask3 = new FloatingTask("Task 3");
+
+		masterList.add(event1);
+		masterList.add(event2);
+		masterList.add(fTask1);
+
+		displayList = new DataLists(masterList);
+
+		assertThat(displayList.getDeadlineTasksList(), IsEqual.equalTo(masterList.getDeadlineTasksList()));
+		assertThat(displayList.getFloatingTasksList(), IsEqual.equalTo(masterList.getFloatingTasksList()));
+		assertThat(displayList.getEventsList(), IsEqual.equalTo(masterList.getEventsList()));
+
+		masterList.add(fTask2);
+		displayList.add(fTask3);
+
+		assertThat(displayList.getFloatingTasksList(), IsNot.not(IsEqual.equalTo(masterList.getFloatingTasksList())));
+	}
+
+	@Test
+	public void sortedEventsTest() {
+		Event event1 = new Event("C Event", LocalDateTime.of(2016, Month.MARCH, 30, 12, 00), 
+				LocalDateTime.of(2016, Month.MARCH, 30, 13, 00));
+		Event event2 = new Event("B Event", LocalDateTime.of(2016, Month.APRIL, 20, 12, 00), 
+				LocalDateTime.of(2016, Month.MARCH, 30, 13, 00));
+		Event event3 = new Event("A Event", LocalDateTime.of(2016, Month.MARCH, 30, 12, 00), 
+				LocalDateTime.of(2016, Month.MARCH, 30, 13, 00));
+		Event event4 = new Event("D Event", LocalDateTime.of(2016, Month.MARCH, 31, 12, 00), 
+				LocalDateTime.of(2016, Month.MARCH, 30, 13, 00));
+
+		masterList.add(event1);
+		masterList.add(event2);
+		masterList.add(event3);
+		masterList.add(event4);
+
+		List<Event> eventsUnsortedList = new ArrayList<Event>();
+
+		eventsUnsortedList.add(event1);
+		eventsUnsortedList.add(event2);
+		eventsUnsortedList.add(event3);
+		eventsUnsortedList.add(event4);
+
+		assertThat(masterList.getEventsList(), IsNot.not(IsEqual.equalTo(eventsUnsortedList)));
+
+		List<Event> eventsSortedList = new ArrayList<Event>();
+
+		eventsSortedList.add(event3);
+		eventsSortedList.add(event1);
+		eventsSortedList.add(event4);
+		eventsSortedList.add(event2);
+
+		assertThat(masterList.getEventsList(), IsEqual.equalTo(eventsSortedList));
+	}
```

Comment:
But if your data was made in a way that that should be sorted with that the correct outcome. Shouldn't it be obvious that that's the order?

I'm testing both the Sorting by time as well as Sorting by name if the time is the same.